.github/workflows/bindings_java.yml

```diff
@@ -70,126 +70,133 @@ jobs:
-  #         MAVEN_GPG_PRIVATE_KEY: op://services/maven/gpg_private_key
-  #         MAVEN_GPG_PASSPHRASE: op://services/maven/gpg_passphrase
-
-  #     - name: Set up JDK 8
-  #       uses: actions/setup-java@v3
-  #       with:
-  #         distribution: 'temurin'
-  #         java-version: '8'
-  #         cache: 'maven'
-  #         server-id: apache.snapshots.https
-  #         server-username: MAVEN_USERNAME
-  #         server-password: MAVEN_CENTRAL_TOKEN
-  #         gpg-private-key: ${{ env.MAVEN_GPG_PRIVATE_KEY }}
-  #         gpg-passphrase: ${{ env.MAVEN_GPG_PASSPHRASE }}
-
-  #     - name: Prepare enviroment variables
-  #       run: echo "LOCAL_STAGING_DIR=$HOME/local-staging" >> $GITHUB_ENV
-
-  #     - name: Download windows staging directory
-  #       uses: actions/download-artifact@v3
-  #       with:
-  #         name: windows-x86_64-local-staging
-  #         path: ~/windows-x86_64-local-staging
-  #     - name: Download linux staging directory
-  #       uses: actions/download-artifact@v3
-  #       with:
-  #         name: linux-x86_64-local-staging
-  #         path: ~/linux-x86_64-local-staging
-  #     - name: Download darwin staging directory
-  #       uses: actions/download-artifact@v3
-  #       with:
-  #         name: osx-x86_64-local-staging
-  #         path: ~/osx-x86_64-local-staging
-  #     - name: Download darwin (aarch64) staging directory
-  #       uses: actions/download-artifact@v3
-  #       with:
-  #         name: osx-aarch_64-local-staging
-  #         path: ~/osx-aarch_64-local-staging
-
-  #     - name: Merge staging repositories
-  #       run: |
-  #         mkdir -p ~/local-staging/deferred
-  #         cat ~/windows-x86_64-local-staging/deferred/.index >>  ~/local-staging/deferred/.index
-  #         cp -r ~/windows-x86_64-local-staging/deferred/* ~/local-staging/deferred/
-  #         cat ~/linux-x86_64-local-staging/deferred/.index >>  ~/local-staging/deferred/.index
-  #         cp -r ~/linux-x86_64-local-staging/deferred/* ~/local-staging/deferred/
-  #         cat ~/osx-x86_64-local-staging/deferred/.index >>  ~/local-staging/deferred/.index
-  #         cp -r ~/osx-x86_64-local-staging/deferred/* ~/local-staging/deferred/
-  #         cat ~/osx-aarch_64-local-staging/deferred/.index >>  ~/local-staging/deferred/.index
-  #         cp -r ~/osx-aarch_64-local-staging/deferred/* ~/local-staging/deferred/
-
-  #     - name: Deploy local staged artifacts
-  #       if: ${{ github.event_name != 'pull_request' }}
-  #       working-directory: bindings/java
-  #       run: mvn org.sonatype.plugins:nexus-staging-maven-plugin:deploy-staged -DaltStagingDirectory=$LOCAL_STAGING_DIR
-  #       env:
-  #         MAVEN_USERNAME: ${{ secrets.NEXUS_USER }}
-  #         MAVEN_CENTRAL_TOKEN: ${{ secrets.NEXUS_PW }}
-  #         MAVEN_GPG_PASSPHRASE: ${{ env.MAVEN_GPG_PASSPHRASE }}
+  stage-snapshot:
+    runs-on: ${{ matrix.os }}
+    if: ${{ github.repository == 'apache/incubator-opendal' }}}
```

Comment:
Please use `github.event_name == 'push' || !github.event.pull_request.head.repo.fork` which is the best query I find to avoid this job runnign on forked repo.